test/js/source/tile_pyramid.test.js

```diff
@@ -648,3 +648,31 @@ test('TilePyramid#orderedIDs (ascending order by zoom level)', function(t) {
     ]);
     t.end();
 });
+
+
+test('TilePyramid#findLoadedParent', function(t) {
+    var pyramid = createPyramid({});
+    var tr = new Transform();
+    tr.width = 512;
+    tr.height = 512;
+    pyramid.updateCacheSize(tr);
+
+    var tile = {
+        coord: new TileCoord(1, 0, 0),
+        loaded: true
+    };
+    pyramid._cache.add(tile.coord.id, tile);
+
+    t.equal(pyramid.findLoadedParent(new TileCoord(2, 3, 3), 0, {}), undefined);
+
+    // add from cache
+    t.deepEqual(pyramid.findLoadedParent(new TileCoord(2, 0, 0), 0, {}), tile);
+
+    // cache is empty
+    t.equal(pyramid._cache.order.length, 0);
+
+    // add from retained tiles
+    t.deepEqual(pyramid.findLoadedParent(new TileCoord(2, 0, 0), 0, {}), tile);
+
+    t.end();
+});
```

Comment:
Thanks for adding unit tests to this method! Can we break this up into a couple separate tests? It looks like you're checking a couple isolate-able bits of functionality.